File: archive/ios-swiftui/RedactPDF/Theme/AppleDesignSystem.swift
import SwiftUI

// MARK: - Apple HIG Design System
// Based on Apple Human Interface Guidelines + Liquid Glass hierarchy principles:
// content sits on solid surfaces; controls float above on translucent materials.

enum AppleDS {

    // MARK: Colors

    enum Color {
        static let canvas = SwiftUI.Color(red: 0x0B / 255, green: 0x0B / 255, blue: 0x0C / 255)
        static let surface = SwiftUI.Color(red: 0x0D / 255, green: 0x0D / 255, blue: 0x0E / 255)
        static let surfaceElevated = SwiftUI.Color(red: 0x14 / 255, green: 0x14 / 255, blue: 0x16 / 255)
        static let surfaceGrouped = SwiftUI.Color(red: 0x1C / 255, green: 0x1C / 255, blue: 0x1E / 255)

        static let labelPrimary = SwiftUI.Color.white
        static let labelSecondary = SwiftUI.Color.white.opacity(0.62)
        static let labelTertiary = SwiftUI.Color.white.opacity(0.42)
        static let labelQuaternary = SwiftUI.Color.white.opacity(0.28)

        static let separator = SwiftUI.Color.white.opacity(0.10)
        static let separatorOpaque = SwiftUI.Color.white.opacity(0.08)

        static let accent = SwiftUI.Color(red: 0.04, green: 0.52, blue: 1.0)
        static let accentMuted = SwiftUI.Color(red: 0.04, green: 0.52, blue: 1.0).opacity(0.18)
        static let success = SwiftUI.Color(red: 0.20, green: 0.84, blue: 0.42)
        static let successMuted = SwiftUI.Color(red: 0.20, green: 0.84, blue: 0.42).opacity(0.16)
    }

    // MARK: Typography (SF Pro semantic scale)

    enum Font {
        static let largeTitle = SwiftUI.Font.system(.largeTitle, design: .rounded, weight: .bold)
        static let title = SwiftUI.Font.system(.title, design: .rounded, weight: .bold)
        static let title2 = SwiftUI.Font.system(.title2, design: .rounded, weight: .semibold)
        static let title3 = SwiftUI.Font.system(.title3, design: .rounded, weight: .semibold)
        static let headline = SwiftUI.Font.system(.headline, design: .default, weight: .semibold)
        static let body = SwiftUI.Font.system(.body, design: .default, weight: .regular)
        static let callout = SwiftUI.Font.system(.callout, design: .default, weight: .regular)
        static let subheadline = SwiftUI.Font.system(.subheadline, design: .default, weight: .regular)
        static let footnote = SwiftUI.Font.system(.footnote, design: .default, weight: .regular)
        static let footnoteMedium = SwiftUI.Font.system(.footnote, design: .default, weight: .medium)
        static let caption = SwiftUI.Font.system(.caption, design: .default, weight: .regular)
        static let captionMedium = SwiftUI.Font.system(.caption, design: .default, weight: .medium)

        static let hero = SwiftUI.Font.system(size: 34, weight: .bold, design: .rounded)
        static let navBrand = SwiftUI.Font.system(size: 20, weight: .bold, design: .rounded)
    }

    // MARK: Spacing (8pt grid)

    enum Spacing {
        static let xxs: CGFloat = 4
        static let xs: CGFloat = 8
        static let sm: CGFloat = 12
        static let md: CGFloat = 16
        static let lg: CGFloat = 20
        static let xl: CGFloat = 24
        static let xxl: CGFloat = 32
        static let xxxl: CGFloat = 40
    }

    // MARK: Radius (continuous corners per HIG)

    enum Radius {
        static let sm: CGFloat = 12
        static let md: CGFloat = 16
        static let lg: CGFloat = 20
        static let xl: CGFloat = 24
        static let pill: CGFloat = 100
        static let icon: CGFloat = 14
    }

    // MARK: Layout

    enum Layout {
        static let minTouchTarget: CGFloat = 44
        static let screenHorizontalPadding: CGFloat = 20
        static let cardPadding: CGFloat = 18
        static let iconContainerSize: CGFloat = 52
    }

    // MARK: Animation

    enum Motion {
        static let spring = Animation.spring(response: 0.38, dampingFraction: 0.82)
        static let springSnappy = Animation.spring(response: 0.28, dampingFraction: 0.86)
        static let easeOut = Animation.easeOut(duration: 0.22)
    }
}

// MARK: - Backward Compatibility

enum AppTheme {
    static let obsidian = AppleDS.Color.surface
    static let dashboardBackground = AppleDS.Color.canvas
    static let subtleBorder = AppleDS.Color.separatorOpaque
    static let cornerRadius = AppleDS.Radius.lg
    static let accentBlue = AppleDS.Color.accent
    static let privacyGreen = AppleDS.Color.success
}

File: archive/flutter/lib/theme/apple_design_system.dart
import 'package:flutter/material.dart';

/// Apple HIG–inspired design tokens for RedactPDF.
/// Content sits on solid surfaces; controls float on translucent materials.
class AppleDS {
  AppleDS._();

  // Colors
  static const Color canvas = Color(0xFF0B0B0C);
  static const Color surface = Color(0xFF0D0D0E);
  static const Color surfaceElevated = Color(0xFF141416);
  static const Color surfaceGrouped = Color(0xFF1C1C1E);

  static const Color labelPrimary = Colors.white;
  static const Color labelSecondary = Color(0x9EFFFFFF);
  static const Color labelTertiary = Color(0x6BFFFFFF);
  static const Color labelQuaternary = Color(0x47FFFFFF);

  static const Color separator = Color(0x1AFFFFFF);
  static const Color separatorOpaque = Color(0x14FFFFFF);

  static const Color accent = Color(0xFF0A85FF);
  static const Color accentMuted = Color(0x2E0A85FF);
  static const Color success = Color(0xFF33D66B);
  static const Color successMuted = Color(0x2933D66B);

  static TextStyle get hero => const TextStyle(
        fontSize: 34,
        fontWeight: FontWeight.w700,
        height: 1.15,
        letterSpacing: -0.5,
        color: labelPrimary,
      );

  static TextStyle get navBrand => const TextStyle(
        fontSize: 20,
        fontWeight: FontWeight.w700,
        color: labelPrimary,
      );

  static TextStyle get title => const TextStyle(
        fontSize: 28,
        fontWeight: FontWeight.w700,
        color: labelPrimary,
      );

  static TextStyle get headline => const TextStyle(
        fontSize: 17,
        fontWeight: FontWeight.w600,
        color: labelPrimary,
      );

  static TextStyle get body => const TextStyle(
        fontSize: 16,
        fontWeight: FontWeight.w400,
        height: 1.4,
        color: labelSecondary,
      );

  static TextStyle get subheadline => const TextStyle(
        fontSize: 15,
        fontWeight: FontWeight.w400,
        color: labelSecondary,
      );

  static TextStyle get footnote => const TextStyle(
        fontSize: 13,
        fontWeight: FontWeight.w400,
        color: labelTertiary,
      );

  static TextStyle get footnoteMedium => const TextStyle(
        fontSize: 13,
        fontWeight: FontWeight.w500,
        color: labelTertiary,
      );

  static TextStyle get caption => const TextStyle(
        fontSize: 12,
        fontWeight: FontWeight.w400,
        color: labelQuaternary,
      );

  static TextStyle get captionMedium => const TextStyle(
        fontSize: 12,
        fontWeight: FontWeight.w600,
        color: labelSecondary,
      );

  static const double xxs = 4;
  static const double xs = 8;
  static const double sm = 12;
  static const double md = 16;
  static const double lg = 20;
  static const double xl = 24;
  static const double xxl = 32;
  static const double xxxl = 40;

  static const double radiusSm = 12;
  static const double radiusMd = 16;
  static const double radiusLg = 20;
  static const double radiusXl = 24;
  static const double radiusIcon = 14;

  static const double minTouchTarget = 44;
  static const double screenPadding = 20;
  static const double cardPadding = 18;
  static const double iconContainer = 52;

  static ThemeData get darkTheme => ThemeData(
        useMaterial3: true,
        brightness: Brightness.dark,
        scaffoldBackgroundColor: canvas,
        colorScheme: const ColorScheme.dark(
          primary: accent,
          surface: surface,
          onSurface: labelPrimary,
        ),
        appBarTheme: const AppBarTheme(
          backgroundColor: canvas,
          foregroundColor: labelPrimary,
          elevation: 0,
          centerTitle: true,
        ),
      );
}
